cli/running/lua/launcher.lua

```diff
@@ -39,22 +39,156 @@ local function check_version(expected)
     return res
 end
 
+-- Types of available options.
+-- Could be comma separated lua types or 'any' if any type is allowed.
+--
+-- get_option_from_env() leans on the set of types in use: don't
+-- forget to update it when add a new type or a combination of
+-- types here.
+local template_cfg = {
```

Comment:
I got it from latest tarantool 1.10.x release. AFAIU getting that list from tarantool 2.x is not the best idea because there could be different behaviours. Of course i can be wrong.

Reply:
> If we'll hardcode it, we'll hit the same problem as cartridge, where the options needs to be updated every time a new option is added into tarantool.
> 
> Can we pre-run tarantool [without snap/xlog file writting](https://github.com/tarantool/tuple-keydef/blob/cd16819f28c1541f28064f6bdafe1206c5304cac/test/tuple_keydef.test.lua#L133-L157) to collect an option list?
> 
> I would also ask for pre-box-cfg API for obtaining all options and/or env variables to be implemented in tarantool in a future.

Why do we have to update the list? This hack is only used for older versions of tarantool, and AFAIK new parameters are not added to older tarantool.

Reply:
It looks like we should update the functionality as Alexander suggested because when mixing fields from versions 1.10 and 2.X (before this was supported by tarantool) we might get a situation where the user sets some unsupported parameters (from 2.x) in 1.10 and `box.cfg()` will fail.